neurom/core/morphology.py

```diff
@@ -453,7 +453,12 @@ def root_node(self):
 
     @cached_property
     def type(self):
-        """The type of the root node."""
+        """The type of the Neurite (which can be composite)."""
+        return NeuriteType(self.subtree_types)
+
+    @cached_property
+    def subtree_types(self):
```

Comment:
I would prefer if the subtrees were not visible from the public api. Why not get them from the type?

Reply:
I am not in favor of extending the api unless there is an essential element for the morphology/neurite. If the user wants to analyze the morphology they can do it manually. subtree processing is also not mature enough to expose it directly to the user.

Reply:
What are you afraid of exactly?
Users can already access to this list of sub-tree types with `Neurite.type.to_list()`. The problem is that when a Neurite object has an unregistered composite type, it's not possible. So in this case it's hard for the users to know what composite type they should register to be able to process this Neurite object.

Reply:
ok, let's revisit this in the future.